packages-exp/auth-types-exp/index.d.ts

```diff
@@ -248,76 +462,202 @@ export abstract class ActionCodeURL {
 }
 
 /**
- * https://firebase.google.com/docs/reference/js/firebase.auth.ApplicationVerifier
+ * A verifier for domain verification and abuse prevention. Currently, the only implementation is
+ * {@link RecaptchaVerifier}.
+ *
+ * @public
  */
 export interface ApplicationVerifier {
+  /** Identifies the type of application verifier (e.g. "recaptcha"). */
   readonly type: string;
+  /**
+   * Executes the verification process.
+   *
+   * @returns A Promise for a token that can be used to assert the validity of a request.
+   */
   verify(): Promise<string>;
 }
 
 /**
- * https://firebase.google.com/docs/reference/js/firebase.auth.RecaptchaVerifier
+ * An {@link https://www.google.com/recaptcha/ | reCAPTCHA}-based application verifier.
+ *
+ * @public
  */
 export abstract class RecaptchaVerifier implements ApplicationVerifier {
   constructor(
+    /**
+     * The reCAPTCHA container parameter. This has different meaning depending on whether the
+     * reCAPTCHA is hidden or visible. For a visible reCAPTCHA the container must be empty. If a
+     * string is used, it has to correspond to an element ID. The corresponding element must also
+     * must be in the DOM at the time of initialization.
+     */
     container: any | string,
+    /**
+     * The optional reCAPTCHA parameters. Check the reCAPTCHA docs for a comprehensive list.
+     * All parameters are accepted except for the sitekey. Firebase Auth backend provisions a
```

Comment:
Is this a literal, and should it be "site key" or "size key?"

We probably should have Kevin Cheung review this, since he has domain knowledge :)